docs/git.rst

```diff
@@ -0,0 +1,304 @@
+.. -*- coding: utf-8 -*-
+
+==================
+git best practices
+==================
+Collection of best practices that would help you and the overall community to work with git more efficiently.
+
+General guidelines
+==================
+
+Pulling code
+------------
+*ALWAYS* (in capital letters) do a :command:`git pull -r` when fetching new code,
+configure git to do always so with::
+
+    git config branch.autosetuprebase always # add the --global switch to make it default everywhere
+
+This way you do not introduce new extra commits and the git history is kept as simple as possible.
+
+This is specially important when trying to understand why some changes were made,
+or who did actually change that line, etc.
+
+Further explanation: http://stevenharman.net/git-pull-with-automatic-rebase
+
+Reviewing your changes
+----------------------
+After hacking for some minutes/hours you are finished and about to commit your changes,
+great!
+
+*BUT* (capital letters),
+please,
+do so with :command:`git add -p`.
+
+The ``-p`` switch allows you to select which hunks you want to add on a commit.
+
+This is not only great to split changes into different commits,
+but is also the time when you actually **review** your code before anyone else sees it.
+
+This is the time when you spot typos,
+pep8 errors,
+misaligned code,
+lack of docstrings in methods...
+
+Remember that the first code review is the one you do on your own.
+Some inspiration/better phrased:
+http://ada.mbecker.cc/2012/11/22/be-your-own-code-review/
+
+And please,
+do remember the gold metric about reviewing code:
+http://www.osnews.com/story/19266/WTFs_m
+
+One commit does one thing
+^^^^^^^^^^^^^^^^^^^^^^^^^
+Repeat with me:
+*One commit does one thing*.
+Period.
+
+When someone else needs to review your code,
+most probably will give up or just skim over your code if that means that you just throw together a commit with +20 files doing all sorts of changes on them.
+
+If you can not express what has been changed within 50 characters
+(suggested length of a commit message subject),
+or you say it like "it does XXX and YYY",
+you most probably need to split that commit into,
+at least,
+two or three commits.
+
+That doesn't mean that a +20 files or +100 lines of code changes are bad per se,
+you may be doing a simple refactoring across lots of files,
+that's fine and good actually,
+as long as that commit is just and only about that refactoring,
+and not a refactoring plus moving things around plus fixing some bugs plus adding some docs plus a new cool feature plus fixing typos on documentation.
```

Comment:
Ironically, this sentence is quite long and could probably be broken down into two shorter sentences. It's a little hard to follow at present.